runtime/program_params_validation_test.go

```diff
@@ -285,6 +285,99 @@ func TestRuntimeScriptParameterTypeValidation(t *testing.T) {
 		assert.NoError(t, err)
 	})
 
+	t.Run("InclusiveRange<Int16>", func(t *testing.T) {
+		t.Parallel()
+
+		script := `
+			pub fun main(arg: InclusiveRange<Int16>) {
+			}
+		`
+
+		err := executeScript(
+			t,
+			script,
+			cadence.NewInclusiveRange(cadence.NewInt16(1), cadence.NewInt16(2), cadence.NewInt16(1)),
+		)
+
+		assert.NoError(t, err)
+	})
+
+	t.Run("InclusiveRange<UInt16> as AnyStruct", func(t *testing.T) {
+		t.Parallel()
+
+		script := `
+			pub fun main(arg: AnyStruct) {
+			}
+		`
+
+		err := executeScript(
+			t,
+			script,
+			cadence.NewInclusiveRange(
+				cadence.NewUInt16(1),
+				cadence.NewUInt16(2),
+				cadence.NewUInt16(1),
+			),
+		)
+
+		assert.NoError(t, err)
+	})
+
+	// Since InclusiveRange isn't covariant.
+	t.Run("Invalid InclusiveRange<Integer>", func(t *testing.T) {
+		t.Parallel()
+
+		script := `
+			pub fun main(arg: InclusiveRange<Integer>) {
```

Comment:
Sorry I forgot some of the details, is it possible to create an `InclusiveRange<Integer>` in cadence?
i.e: Is following allowed?
```cadence
let a: Integer = 0
let b: Integer = 10
var range: InclusiveRange<Integer> = InclusiveRange(a, b)
```

Reply:
Good question.

Yes, the code works but I think it should not work. 

I disallowed it during import by checking it in `InclusiveRangeStaticTypeConformsToStaticType` (Called by `ConformsToStaticType` for `CompositeValue`)

I think I can do the same in the InclusiveRange constructor as well. WDYT?

Reply:
It seems not just the constructor, the checker also allows covariant subtyping:
e.g:
```cadence
let a: InclusiveRange<Int> = InclusiveRange(0, 10)
let b: InclusiveRange<Integer> = a
```
This would need to be fixed in the `IsSubType()` method, which would fix it universally (won't need to special case in multiple places)

Reply:
While testing, I also ran into the following cases that currently report errors, but shouldn't:

1) The inclusive range constructor always takes `Int` as the type parameter's type.
```cadence
let r = InclusiveRange<Int>(0, 10)    // Works

let r = InclusiveRange<Int8>(0, 10)    // Checker error (should also work)
```

2) Also probably related to above, Type inferring also doesn't infer the type parameter from LHS.
```cadence
let b: InclusiveRange<Int8> = InclusiveRange(0, 10)   // Type-mismatch error
```